### PR TITLE
Add test for degradation with threads

### PR DESCRIPTION
**Pull request by dcristoloveanu**
- Add a test that reproduces the problem of perf degradation with number of threads being used

### CODE REVIEW

File: build/devops_gated.yml [hunk 1]
```diff
@@ -311,6 +311,13 @@ jobs:
     workingDirectory: 'build_x64'
     displayName: 'Run lock_free_set_perf tests'
 
+  - task: CmdLine@1
+    displayName: 'Run ctest'
```
Comment (mattdurak) [inline build/devops_gated.yml:315]: >un ctest' [](http://example.com/codeflow?start=19&length=9)

Should this say "Run ctests - Perf" or "Run perf tests"?

**Review by mattdurak — APPROVED** [review]
:shipit:

File: build/devops_gated.yml (hunk 1 above)
Comment (dcristoloveanu) [reply build/devops_gated.yml]: Done.

**Review by mattdurak — APPROVED** [review]
:shipit:

**Pull request merged**